Bug fixes and small enhancements

gibz104 opened the pull request:
**Summary**

This PR adds a settings UI for customization options and fixes the QR code scanning functionality which was not working.

**Changes:**

1. Settings UI
- Added a settings dialog (gear icon in toolbar) with:
  - Accent color picker with 8 preset colors + custom color option
  - Checkbox to show spool IDs in dropdowns (addresses #19)
- Settings persist to localStorage
- Accent color applies to toolbar and buttons via CSS custom properties

2. QR Code Scanner Fix
- Problem: The QR scanner was completely non-functional with Spoolman's QR codes. It was passing the raw QR code value (e.g., web+spoolman:s-4 or https://.../spool/show/4) directly to a barcode lookup API, expecting an exact match in the spool's barcode field. This would never work with Spoolman's built-in QR codes.
- Fix: Added proper parsing for Spoolman QR code formats to extract the spool ID, then look up the spool by ID from the existing spools list. Supports:
  - web+spoolman:s-{id} format
  - https://.../spool/show/{id} URL format
  - Plain numeric IDs
  - Falls back to barcode field matching for custom barcodes

3. Error Handling on Scan Page
- Problem: When the spool lookup failed, the page showed no feedback.  Clicking tray buttons did nothing with no error message.
- Fix: Added loading states, error messages, and a "Go Home" button when errors occur. Users now see clear feedback like "No spool found for barcode: xyz".

4. Spool ID Display
- When "Show Spool IDs" is enabled, IDs appear at the beginning of dropdown entries (`#4 Bambu Lab PETG...`) so they don't get cut off
- Spool ID shown subtly at bottom of spool info cards

**Testing:**

- Tested QR scanning with both Spoolman QR code formats
- Tested accent color theming
- Tested spool ID display toggle

**Screenshots:**
![Demo](https://github.com/user-attachments/assets/0d67f133-8e90-4125-a0e3-dcd533144d9c)

<img width="1058" height="532" alt="image" src="https://github.com/user-attachments/assets/5ca18e22-21f6-48e8-b175-e4837b7ad840" />

gibz104 committed:
Add settings UI and fix QR code scanner

gibz104 committed:
Remove unused /Spools/barcode endpoint

gibz104 commented:
I've also removed the unused `/Spools/barcode` backend endpoint.  Barcode lookups are now handled client-side by filtering the spool list returned from `GET /Spools`, which already contains the barcode field.  This simplifies the codebase by eliminating redundant server-side lookup logic

gibz104 committed:
Fix external spool bug

gibz104 commented:
I've also included a bug fix for the external spool not persisting the selected spool as discussed in #20

marcokreeft87 commented:
> I've also removed the unused `/Spools/barcode` backend endpoint. Barcode lookups are now handled client-side by filtering the spool list returned from `GET /Spools`, which already contains the barcode field. This simplifies the codebase by eliminating redundant server-side lookup logic

The barcode endpoint is being used. Maybe not in your changes, but I scanned the barcode with the camera button and than it selected the spool I scanned. 

Can you alter your change to only add functionality for now?
Not everyone wants to print qr codes every time they get a new spool

gibz104 commented:
> > I've also removed the unused `/Spools/barcode` backend endpoint. Barcode lookups are now handled client-side by filtering the spool list returned from `GET /Spools`, which already contains the barcode field. This simplifies the codebase by eliminating redundant server-side lookup logic
> 
> The barcode endpoint is being used. Maybe not in your changes, but I scanned the barcode with the camera button and than it selected the spool I scanned.
> 
> Can you alter your change to only add functionality for now? Not everyone wants to print qr codes every time they get a new spool

My changes do not require users to print QR codes every time.  The end user functionality of using barcodes is still intact.  I've just also added the ability to use Spoolman QR codes in addition to barcodes.  The change I made was purely on the backend and a simplification of existing logic.  We already get the barcode field from spoolman when calling GET /spools, so the /Spools/barcode endpoint was unnecessary.  I've tested on my version of the code and both barcodes (from the Spoolman extra data field) and Spoolman QR codes work.  Therefore, end user functionality was not changed and only added to.
